source/samples/pubsub/PubSubFeature.cpp

```diff
@@ -30,6 +34,54 @@ string PubSubFeature::getName()
     return "Pub Sub Sample";
 }
 
+bool PubSubFeature::createPubSub(const PlainConfig &config, std::string absFilePath)
+{
+    std::string pubFileDir = FileUtils::ExtractParentDirectory(absFilePath);
+    if (!FileUtils::DirectoryExists(pubFileDir))
+    {
+        // Create an empty directory with the expected permissions.
+        FileUtils::CreateDirectoryWithPermissions(pubFileDir.c_str(), S_IRWXU | S_IRGRP | S_IROTH | S_IXOTH);
+    }
+    else
+    {
+        // Verify the directory permissions.
+        auto rcvDirPermissions = FileUtils::GetFilePermissions(pubFileDir);
+        if (Permissions::PUBSUB_DIR != rcvDirPermissions)
+        {
+            LOGM_ERROR(
+                TAG,
+                "Incorrect directory permissions for pubsub file: %s expected: %d received: %d",
+                Sanitize(pubFileDir).c_str(),
+                Permissions::PUBSUB_DIR,
+                rcvDirPermissions);
+            return false;
+        }
+    }
+
+    if (!FileUtils::FileExists(absFilePath))
+    {
+        // Create an empty file with the expected permissions.
+        FileUtils::CreateEmptyFileWithPermissions(absFilePath, S_IRUSR | S_IWUSR);
+    }
+    else
+    {
+        // Verify the file permissions.
+        auto rcvFilePermissions = FileUtils::GetFilePermissions(absFilePath);
+        if (Permissions::PUB_SUB_FILES != rcvFilePermissions)
+        {
+            LOGM_ERROR(
+                TAG,
+                "Incorrect file permissions for pubsub file: %s expected: %d received: %d",
+                Sanitize(absFilePath).c_str(),
+                Permissions::PUB_SUB_FILES,
+                rcvFilePermissions);
+            return false;
+        }
+    }
+
```

Comment:
[Minor] Suggest to include a log message at INFO that prints pubsub file path.

Reply:
done